src/main/scala/lms/core/stub.scala

```diff
@@ -106,6 +106,13 @@ object Adapter extends FrontEnd {
           curEffects.get(as).filter({ case (_, lrs) => lrs contains rs.asInstanceOf[Sym] }).isDefined } =>
         Some(Const(()))
 
+      case ("array_set", List(as: Exp, i, rs)) if ({
```

Comment:
Can you add a comment that show the optimization as we have above.

// as(i) = rs; ...; as(i) = rs => as(i) = rs; ...; ()   side condition no brite in-between

Reply:
will do